### PR TITLE
OSDOCS-10894:adds troubleshooting to ANP

### PR DESCRIPTION
**Pull request by JoeAldinger**
Version(s):
4.16

Issue:
https://issues.redhat.com/browse/OSDOCS-10894

Link to docs preview:

QE review:
- [x] QE has approved this change.

Additional information:
PTAL @asood-rh cc @tssurya

### CODE REVIEW

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-troubeshooting.adoc:32]: You suggest using `kubectl get` but the output on the example is from `kubectl describe`. It should be consistent (either provide some YAML output or change the suggestion to use describe)

In addition, it might be best to suggest using `oc` instead of `kubectl`. I know that `oc` is "`kubectl` plus more things" and that it behaves as `kubectl` if invoked as such, but for OCP documentation, it may be best to stick to use `oc` in the examples (in upstream docs, on the other hand, people have to use `kubectl` because `oc` is OCP-specific, but that's not our case here)

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:8]: Shouldn't we include also how to get the port groups, as in the [upstream docs](https://github.com/ovn-org/ovn-kubernetes/pull/4431)?

cc @tssurya

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:159]: I'd remove step 1 and make this one non-optional, i.e. just ask the customer to always provide the output with node information.

Reason: Most of the times you troubleshoot something, you do it because you have problems on a node and you look for the ovnkube-node-XXXX pod of that node, so 99% of the times you want the pods with node information, so we can also keep the doc a bit shorter.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:191]: Listing ACLs on a big node (or, better said, in a node which is not small) can be a too big output. Most of the times you want to grep (or use `ovn-nbctl find` like in the latest [comments I made on the upstream docs](https://github.com/ovn-org/ovn-kubernetes/pull/4431#discussion_r1644619505)

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:385]: Same than what I said for listing all the ACLs.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (tssurya) [reply modules/nw-ovn-k-anp-nbctl.adoc:8]: yes we must

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (tssurya) [reply modules/nw-ovn-k-anp-nbctl.adoc:191]: agreed :) list will be noise

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:463]: As [commented upstream](https://github.com/ovn-org/ovn-kubernetes/pull/4431#discussion_r1644619505), using `ovn-nbctl find` like in the comment might work better. But it might be wise if we do the same than the upstream docs does, for consistency.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:470]: I also made another [suggestion upstream](https://github.com/ovn-org/ovn-kubernetes/pull/4431/files#r1646486961) regarding this.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:497]: This output would be possible only if you removed `grep` as per my suggestion from upstream above. With grep, you would need some `-A`, `-B` or `-C` parameters to get this output.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (tssurya) [reply modules/nw-ovn-k-anp-nbctl.adoc]: ugh stale review, dismiss this comment

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:32]: I changed this to `oc describe anp` and `oc describe banp` that's what I used. I think this was just a mixup between writing from upstream and working on a live cluster.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:191]: agreed, I'll add.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:159]: done

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [inline modules/nw-ovn-k-anp-nbctl.adoc:175]: Doing another round of review, I see that this element uses example names while the next steps uses placeholders like `<anp_name>`. It would be best to have this consistent.

So it is best to replace `cluster-control` with `<anp_name>` and `AdminNetworkPolicy` with a `<kind>` placeholder.

Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:175]: updated

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-troubeshooting.adoc:10]: Just a wording suggestion
```suggestion
A good status indicates `OVN DB plumbing was successful` and the `SetupSucceeded`.
```

Comment (opayne1) [inline modules/nw-ovn-k-anp-troubeshooting.adoc:10]:
```suggestion
A good status will indicate that `OVN DB plumbing was successful` and that `SetupSucceeded`.
```

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-troubeshooting.adoc:34]: I dont think you need "you will see"
```suggestion
If plumbing is unsuccessful, an error is reported from the respective zone controller on the status.
```

Comment (opayne1) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:34]: Also, follow up question is the status just saying there was an error? If so I think you could leave that part out. So,
```suggestion
If plumbing is unsuccessful, an error is reported from the respective zone controller.
```

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-troubeshooting.adoc:36]: Do you think it would be useful to name the following section? I don't recall seeing something like this in the docs, but I also haven't explicitly looked for it.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-nbctl.adoc:8]:
```suggestion
To troubleshoot an unsuccessful setup, start by looking at OVN Northbound database (nbdb) objects including `ACL`, `AdressSet`, and `Port_Group`. To view the nbdb, you need to be inside the pod on that node to see what objects are in the node.
```
The second sentence reads funky to me because you tell users to "view" the ndbd, but then to "see" what objects are in the node. Seems that only one "view" or "see" is needed, but I am not sure how to rephrase. I'll leave that to you. It is also fine as is, I was just thinking.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-nbctl.adoc:17]:
```suggestion
To run ovn `nbctl` commands in a cluster, you must open a remote shell into the `nbdb` on the relevant node.
```
I think a comma should go there. Also should it be OVN `nbctl` or `ovn nbctl`?

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (opayne1) [inline modules/nw-ovn-k-anp-nbctl.adoc:347]: Is there supposed to be something in the quotations after "see"?

**Review by opayne1 — COMMENTED** [review]
Looks great! Just a few nits and I had a few small suggestions. Let me know if you have any questions on them!

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (asood-rh) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:36]: The commands are actually ovn-nbctl
E.g.
<pre>
oc -n openshift-ovn-kubernetes exec -it ovnkube-node-fqgqn -- ovn-nbctl lr-route-list GR_openshift-qe-027.sriov.openshift-qe.sdn.com
</pre>
@JoeAldinger can use this format where you exec into pod and issue command instead of rsh into pod.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:8]: missing the word database.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:347]: nope

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:10]: I like this.

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:36]: @asood-rh I was trying to keep consistency with other procedures in the docs: https://docs.openshift.com/container-platform/4.15/networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture-assembly.html#nw-ovn-kubernetes-list-database-contents_ovn-kubernetes-architecture:~:text=the%20relevant%20node-,List,-pods%20by%20running

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (asood-rh) [inline modules/nw-ovn-k-anp-nbctl.adoc:174]: Alternatively use the form
<pre>
oc rsh -n openshift-ovn-kubernetes exec -it ovnkube-node-524dt -- ovn-nbctl find ACL 'external_ids{>=}{"k8s.ovn.org/owner-type"=AdminNetworkPolicy,"k8s.ovn.org/name"=cluster-control}'
</pre>

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (asood-rh) [inline modules/nw-ovn-k-anp-nbctl.adoc:369]: Example mismatch. The ACL below does not look like for Egress rule

**Review by palonsoro — CHANGES_REQUESTED** [review]
I'd only make some small comments.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:168]: We don't do NBDB; we do nbdb.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:369]: @palonsoro

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:174]: since we use the ovn-nbctl in the netpol I'm just going to keep this as is. I do understand that this works as well.

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (palonsoro) [reply modules/nw-ovn-k-anp-nbctl.adoc:369]: It is. Can you please adjust the command to be like this?
```
ovn-nbctl find ACL 'external_ids{>=}{"k8s.ovn.org/owner-type"=AdminNetworkPolicy,direction=Ingress,"k8s.ovn.org/name"=cluster-control,gress-index="1"}'
```

and also adjust the explanation?

Comment (palonsoro) [reply modules/nw-ovn-k-anp-nbctl.adoc:369]: It now looks correct

File: modules/nw-ovn-k-anp-troubeshooting.adoc
Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:34]: I'll put this in. I was waiting for my cluster to spin up and you beat me to it.

Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-troubeshooting.adoc:37]: Done

File: modules/nw-ovn-k-anp-nbctl.adoc
Comment (asood-rh) [inline modules/nw-ovn-k-anp-nbctl.adoc:161]: Why this specific ovnkube-node pod?

Comment (JoeAldinger) [reply modules/nw-ovn-k-anp-nbctl.adoc:161]: No reason, I just picked one at random

Comment (asood-rh) [reply modules/nw-ovn-k-anp-nbctl.adoc:161]: it is OK to pick any ovnkube-node, checked it on live cluster.

**Pull request merged**
